### PR TITLE
Authentication feature

### PR DESCRIPTION
**Pull request by jeinygroove**
# Authentication feature

Now user needs to authenticate if he wants to add comments or vote for them (I still don't restrict voting, so user can vote as much as he want at this moment). Also user can delete his own comments only.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comments.java [hunk 1]
```diff
@@ -109,6 +119,20 @@ public void deleteComment(long id) {
         this.datastore.delete(commentEntityKey);
     }
 
+    /**
+     * Checks if comment with given id exists or not.
+     * @param id    Id of the "Comment" entity
+     * @return true, if comment exists, and false if not
+     */
+    public boolean isCommentExist(long id) {
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/data/Comments.java]: nit `commentExists` is a better name... "is comment exist" doesn't make a lot of sense when you read it

File: portfolio/src/main/java/com/google/sps/data/UserManager.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+public class UserManager {
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/data/UserManager.java:11]: Please add some documentation to this class

File: portfolio/src/main/java/com/google/sps/data/UserManager.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    public static String getCurrentUserId() {
+        UserService userService = UserServiceFactory.getUserService();
+
+        if (!userService.isUserLoggedIn()) {
+            return null;
+        }
+
+        return userService.getCurrentUser().getUserId();
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/data/UserManager.java]: Optionally you could return a ternary here.

`return userService.isUserLoggedIn() ? userService.getCurrentUser().getUserId() : null;`

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 4]
```diff
@@ -17,9 +17,12 @@
 import com.google.appengine.repackaged.com.google.common.annotations.VisibleForTesting;
 import com.google.sps.data.Comments;
 import com.google.gson.Gson;
+import com.google.sps.data.UserManager;
+
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: nit: remove empty space between imports

File: portfolio/src/main/webapp/js/init.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Authenticates the user.
+ */
```
Comment (amaurymedeiros) [inline portfolio/src/main/webapp/js/init.js]: nit: Authenticates or log the user out as well, right?

**Review by amaurymedeiros — APPROVED** [review]
Only minor comments

File: portfolio/src/main/webapp/js/init.js (hunk 5 above)
Comment (jeinygroove) [reply portfolio/src/main/webapp/js/init.js]: Not really, it checks if the user is logged in, if yes, it returns a link for logging out, if no, a link for signing in. I think there should be "Checks if current user is logged in" then..

**Pull request merged**